src/Stryker.Core/Stryker.Core/TestRunners/VsTest/VsTestRunner.cs

```diff
@@ -401,7 +401,15 @@ private void Dispose(bool disposing)
             {
                 if (disposing)
                 {
-                    _vsTestConsole.EndSession();
+                    try
+                    {
+                        _vsTestConsole.EndSession();
```

Comment:
Perhaps the VsTestRunnerPool should check whether the VsTestRunner agent is still connected before releasing the agent from the pool and if it is not connected it should remove the disconnected agent and spin up a new one. I believe vsTestConsole exposes the connection status to the consumer so we can pass that through to the pool.